### PR TITLE
Correct pluralisation of various diagnostic messages

### PR DESCRIPTION
**Pull request by varkor**

### CODE REVIEW

File: src/librustc/ty/error.rs [hunk 1]
```diff
@@ -200,7 +200,9 @@ impl<'tcx> ty::TyS<'tcx> {
             ty::Array(_, n) => {
                 let n = tcx.lift_to_global(&n).unwrap();
                 match n.try_eval_usize(tcx, ty::ParamEnv::empty()) {
-                    Some(n) => format!("array of {} elements", n).into(),
+                    Some(n) => {
+                        format!("array of {} element{}", n, if n != 1 { "s" } else { "" }).into()
```
Comment (Centril) [inline src/librustc/ty/error.rs]: This `if $n != 1 { "s" } else { "" }` business should be refactored to a function in some utility module in `rustc_errors`. Would be good to grep for `"s" } else { ""`.

Comment (varkor) [reply src/librustc/ty/error.rs]: That would make a nice beginner issue. I'll open one.

Comment (varkor) [reply src/librustc/ty/error.rs]: Opened https://github.com/rust-lang/rust/issues/64238.

**Pull request merged**